services/galley/src/Galley/API/MLS/Welcome.hs

```diff
@@ -102,5 +108,17 @@ sendLocalWelcomes con now rawWelcome lclients = do
           e = Event (qUntagged lcnv) (qUntagged lusr) now $ EdMLSWelcome rawWelcome
        in newMessagePush lclients () (Just con) defMessageMetadata (u, c) e
 
-sendRemoteWelcomes :: ByteString -> Remote [(UserId, ClientId)] -> Sem r ()
-sendRemoteWelcomes = undefined
+sendRemoteWelcomes ::
+  Members '[FederatorAccess] r =>
+  ByteString ->
+  Remote [(UserId, ClientId)] ->
+  Sem r ()
+sendRemoteWelcomes rawWelcome rClients = do
+  let req =
+        MLSWelcomeRequest
+          { mwrRawWelcome = Base64ByteString rawWelcome,
+            mwrRecipients = MLSWelcomeRecipient <$> tUnqualified rClients
+          }
+  let rpc = fedClient @'Galley @"mls-welcome" req
+  -- TODO(md): do error handling
```

Comment:
@mdimjasevic what errors should we handle here?

Reply:
Now that I think about it, I don't see any left. We handle errors in dereferencing a key package reference even before getting into this function, so I think there's no errors left. So feel free to drop that note unless you see any error cases we've forgotten to handle.